torch/functional.py

```diff
@@ -324,6 +325,118 @@ def parse_subscript(n: int) -> str:
 
     return _VF.einsum(equation, operands)  # type: ignore[attr-defined]
 
+# If pad specifier is not None and not a tensor, convert it to a tensor
+def _maybe_convert_pad_specifier(pad_spec, arg_name, dtype, device):
+    if pad_spec is None or torch.is_tensor(pad_spec):
+        return pad_spec
+
+    # TODO: torch.tensor(), when called outside of pytorch, throws a warning
```

Comment:
Maybe there's a way to get warnings back by setting the warning filter? But barring that this should maybe reduce to just a one-line TODO

Reply:
I tried this again and a few of the CI jobs had issues with it. I'm not sure why, maybe float-to-int warnings can be environment dependent? I've reduced the TODO comment.